### PR TITLE
updated reamdme with how to add permission groups to a user

### PR DESCRIPTION
**Pull request by abroddrick**
## Ticket

Resolves action item from Eng huddle

All other changes require just a single approving review.-->

## Changes

- updated Readme with the django admin instructions for how to add user groups (and the connected permissions) to a user.

## Context for reviewers
This will be the instructions that need to be followed on stable where fixtures will NOT be run for new employees etc. Users that want special permission (such as new employees, testers etc) will need to be granted permissions via django admin and not code. Please make sure these instructions make sense for anyone to follow, with the assumption being they at least know what django-admin is. 

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: docs/django-admin/roles.md [hunk 1]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
 
 For more details, refer to the [user group model](../../src/registrar/models/user_group.py).
 
+## Adding a user as analyst or granting full access via django-admin
+
+If a new team member has joined whose data is NOT included in fixtures,
+then they will need to be granted analyst ( `cisa_analysts_group`) or full access (`full_access_group`) permissions in order to view the django-admin pages.
```
Comment (neilmb) [inline docs/django-admin/roles.md:15]: **question**: I think that a `User` object also needs to be marked `is_staff` to be able to log in to admin pages. Can we add that to these instructions?

File: docs/django-admin/roles.md [hunk 2]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
...
+5. After clicking the right arrow in the previous step, scroll down and click `Save`
+
+## Removing a user group permission via django-admin
+
+If an employee was given the wrong permissions or has had a change in roles that subesequently requires a permission change, then their permissions should be updated in django-admin. Much like in the previous section you can accomplish this by doing the following:
+
+1. Go to the `Users` table an select the username for the user in question
+2. Scroll down to the `Permissions` section and find the `Groups` table which has a column for `Available groups` and `Chosen groups`.
+3. In this table, select the permission you want to remove from the `Chosen groups` and then click the left facing arrow to move the permission to `Available groups`.
+4. Depending on the scenario you may now need to add the opposite permission group to the `Chosen groups` section, please see the above section for instructions on how to do that.
```
Comment (neilmb) [inline docs/django-admin/roles.md:31]: **question**: Similarly, I think that we need to remove the `is_staff` property of a user who should no longer be able to see the admin at all.

**Review by neilmb — COMMENTED** [review]
Great content addition, but one question about `is_staff`.

File: docs/django-admin/roles.md [hunk 3]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
...
+
+1. The user in question will need to have a login.gov account and login into our system, this will create a `Users` table entry with their email address and name.
+2. On that `Users` table note that the `GROUP` column should be blank for them as they have no special permisisons yet.
+3. Click on their username, then scroll down to the `Permissions` section.
+4. Under `Permissions`, see the `Groups` table which has a column for `Available groups` and `Chosen groups`. Select the permission you want from the `Available groups` column and click the right arrow to move it to the  `Chosen groups`. Note, if you want this user to be an analyst select `cisa_analysts_group`, otherwise select the `full_access_group`.
+5. After clicking the right arrow in the previous step, scroll down and click `Save`
+
+## Removing a user group permission via django-admin
+
+If an employee was given the wrong permissions or has had a change in roles that subesequently requires a permission change, then their permissions should be updated in django-admin. Much like in the previous section you can accomplish this by doing the following:
```
Comment (zandercymatics) [inline docs/django-admin/roles.md:26]:
```suggestion
If an employee was given the wrong permissions or has had a change in roles that subsequently requires a permission change, then their permissions should be updated in django-admin. Much like in the previous section you can accomplish this by doing the following:
```
Minor typo

File: docs/django-admin/roles.md [hunk 4]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
...
 For more details, refer to the [user group model](../../src/registrar/models/user_group.py).
 
+## Adding a user as analyst or granting full access via django-admin
+
+If a new team member has joined whose data is NOT included in fixtures,
+then they will need to be granted analyst ( `cisa_analysts_group`) or full access (`full_access_group`) permissions in order to view the django-admin pages.
+To do this, do the following:
+
+1. The user in question will need to have a login.gov account and login into our system, this will create a `Users` table entry with their email address and name.
+2. On that `Users` table note that the `GROUP` column should be blank for them as they have no special permisisons yet.
```
Comment (zandercymatics) [inline docs/django-admin/roles.md:19]:
```suggestion
2. On that `Users` table note that the `GROUP` column should be blank for them as they have no special permissions yet.
```

File: docs/django-admin/roles.md [hunk 5]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
...
+3. Click on their username, then scroll down to the `Permissions` section.
+4. Under `Permissions`, see the `Groups` table which has a column for `Available groups` and `Chosen groups`. Select the permission you want from the `Available groups` column and click the right arrow to move it to the  `Chosen groups`. Note, if you want this user to be an analyst select `cisa_analysts_group`, otherwise select the `full_access_group`.
```
Comment (zandercymatics) [inline docs/django-admin/roles.md:21]: This step could benefit from a screenshot (could just be the page they should do this on) or link to a loom just to help orient somebody who may be lost at this stage. I think its fairly intuitive but usually a visual reference for chained steps helps orient people faster. 

If a screenshot would look weird here and/or goes against style rules, splitting this up into separate steps would also help and I think would work just as well.

File: docs/django-admin/roles.md (hunk 2 above)
Comment (zandercymatics) [inline docs/django-admin/roles.md:31]: Ah it is split up! Nice!

Comment (zandercymatics) [inline docs/django-admin/roles.md:31]:
```suggestion
1. Go to the `Users` table an select the username for the user in question
2. Scroll down to the `Permissions` section and find the `Groups` table which has a column for `Available groups` and `Chosen groups`.
3. In this table, select the permission you want to remove from the `Chosen groups` and then click the left facing arrow to move the permission to `Available groups`.
4. Depending on the scenario you may now need to add the opposite permission group to the `Chosen groups` section, please see the section above for instructions on how to do that.
```
(Optional) Very nitpicky but I think this reads a bit better

**Review by zandercymatics — APPROVED** [review]
Great documentation! Love to see this here, its super helpful to have this written down. 

Found a few typos, but otherwise pretty straight forward. The only thing that I think needs a second look is step 4 of `## Adding a user as analyst or granting full access via django-admin` which could benefit from being split up into constituent steps like it was done on the second section, and/or adding a visual reference if you feel like it'd help. 

Nothing is blocking though. Everything minus the typo fixes is optional here, great work!

File: docs/django-admin/roles.md (hunk 5 above)
Comment (abroddrick) [reply docs/django-admin/roles.md:21]: Yeah unforntunetly the screen shot wouldn't work great here but agreed it would be nice to include one

File: docs/django-admin/roles.md (hunk 1 above)
Comment (abroddrick) [reply docs/django-admin/roles.md:15]: Good call out, just DM-ed rachid to confirm this, but if this is the case, it definitely shows why a readme is needed, I didn't realize we still had to add is_staff and assumed that would be covered under permissions.

Comment (therealslimhsiehdy) [reply docs/django-admin/roles.md:15]: **nit:** TOTALLY an extreme nitpick, but removing the space between ( and `:

( `cisa_analysts_group`) -> (`cisa_analysts_group`)

File: docs/django-admin/roles.md [hunk 6]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
...
 
+## Adding a user as analyst or granting full access via django-admin
+
+If a new team member has joined whose data is NOT included in fixtures,
+then they will need to be granted analyst ( `cisa_analysts_group`) or full access (`full_access_group`) permissions in order to view the django-admin pages.
+To do this, do the following:
+
+1. The user in question will need to have a login.gov account and login into our system, this will create a `Users` table entry with their email address and name.
+2. On that `Users` table note that the `GROUP` column should be blank for them as they have no special permissions yet.
+3. Click on their username, then scroll down to the `Permissions` section.
```
Comment (therealslimhsiehdy) [inline docs/django-admin/roles.md:20]: **nit:** `Permissions` -> `User Permissions` (and would be same for in step 4 too if you decide to change and in step 2 for removing section)

(A total extreme nitpick, and feel free to ignore these if it's not useful --  I'm just following along with the instructions and double checking and dunno if it was important making sure we had the exact section name 😅)

File: docs/django-admin/roles.md [hunk 7]
```diff
@@ -9,6 +9,28 @@ our `user_group` model and run in a migration.
 
 For more details, refer to the [user group model](../../src/registrar/models/user_group.py).
 
+## Adding a user as analyst or granting full access via django-admin
+
+If a new team member has joined whose data is NOT included in fixtures,
```
Comment (therealslimhsiehdy) [inline docs/django-admin/roles.md:14]: **Dumb question:** Would we want to add a link of where to go to make these changes (ie getgov-ENV.app.cloud.gov/admin)

(I'm assuming that this is doc is not just for dev use so just in case, but feel free to ignore)

**Review by therealslimhsiehdy — APPROVED** [review]
Just left some nits that you can 100% ignore -- looks great, and thank you so much for putting this all together!

**Comment by abroddrick** [comment]
@h-m-f-t before I merge this, will this make sense to analysts and do you want it copied into a google doc or something for ease of accessing?

File: docs/django-admin/roles.md (hunk 5 above)
Comment (h-m-f-t) [reply docs/django-admin/roles.md:21]: Why couldn't a screenshot be included in this Markdown file?

File: docs/django-admin/roles.md [hunk 8]
```diff
@@ -9,6 +9,30 @@ our `user_group` model and run in a migration.
...
```
Comment (h-m-f-t) [inline docs/django-admin/roles.md:14]: I don't think analysts will know what fixtures is or what it matters. Can a short description and/or link be added?

**Review by h-m-f-t — COMMENTED** [review]
Thanks @abroddrick! I left a few (non-blocking) comments. Adding documentation here (over Drive, etc.) is fine.

File: docs/django-admin/roles.md (hunk 5 above)
Comment (zandercymatics) [reply docs/django-admin/roles.md:21]: @abroddrick Its not blocking at all, but I think the main thing for this is not a screenshot but that step 4 needs additional context or to be kind of "spaced out" if that makes sense. Doing so would help it be easier to follow, though its not necessarily hard to follow now (in other words, its nitpicky). For an example, I like how the instructions are split up in another section of your code, below: 
image
